talentsearch/resources/js/components/skills/SkillChecklist.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { ChangeEvent } from "react";
+import groupBy from "lodash/groupBy";
+import { useIntl } from "react-intl";
+import { getLocale } from "@common/helpers/localize";
+import { SkillFamily } from "@common/api/generated";
+import {
+  getSkillCategoryText,
+  getSkillCategoryImage,
+} from "@common/constants/localizedConstants";
+
+interface FamilyProps {
+  family: SkillFamily;
+  index: string;
+  checked: boolean;
+  callback: (event: ChangeEvent<HTMLInputElement>) => void;
```

Comment:
This works, but to be honest, I don't usually pass the whole event into these kinds of handlers. There are times when its useful, but most of the time you just want to know the data associated with this component. You can make that really explicit with a function definition like `(family: SkillFamily, checked: boolean) => void`. That way, the parent component defining the handler function doesn't need to worry about which piece of data was attached to the input element: was it id? name? index? 
If the Family component just calls `callback(family, e.target.checked)` then its parent knows explicitly what happened, without coupling the definition of the handler function to the details of the input element. Explicit is good, and lack of coupling is good.

Reply:
Yeah that's a much better way of doing things.